server/api.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func (p *Plugin) SetupRouting() {
+	root := mux.NewRouter()
+	api := root.PathPrefix("/api/v0").Subrouter()
+	api.Use(MattermostAuthorizationRequired)
+	api.Handle("{anything:.*}", http.NotFoundHandler())
+	api.NotFoundHandler = http.NotFoundHandler()
+
+	note := api.PathPrefix("/note/{channelid:[A-Za-z0-9]+}").Subrouter()
+	note.Use(p.PermissionsToChannel)
+	note.HandleFunc("", p.getNote).Methods(http.MethodGet)
+	note.HandleFunc("", p.saveNote).Methods(http.MethodPost)
+
+	p.root = root
+}
+
+type SaveNoteRequest struct {
+	Note string `json:"note"`
+}
+
+func (p *Plugin) saveNote(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	channelId := vars["channelid"]
+
+	var noteRequest SaveNoteRequest
+	if err := json.NewDecoder(r.Body).Decode(&noteRequest); err != nil {
+		HandleErrorWithCode(w, http.StatusBadRequest, "failed to decode note request", err)
+		return
+	}
+
+	p.API.KVSet("note_"+channelId, []byte(noteRequest.Note))
```

Comment:
You can only save one note per channel? Or it's a like a collaborative "notes" field? I guess I forget what the UX is supposed to be (0/5 maybe put a gif recording for the PR review process?)

Reply:
Exactly, one note per channel. The README has some pictures.